functorch/experimental/_nonzero_static.py

```diff
@@ -0,0 +1,65 @@
+import torch
+import torch.library
+
+
+"""
+Define a `nonzero_static` operation.
+"""
+
+
+def nonzero_static(self: torch.Tensor, *, size: int, fill_value=-1):
+    """
+    Returns a 2-D tensor where each row is the index for a nonzero value.
+
+    The number of rows will alway be `size`.
+    If the actual number of non-zero elements is smaller than `size`, the return
+    value will be padded with 'fill_value'.
+    If the actual number of non-zero elements is larger than `size`, the return
+    value will be truncated.
+
+    For example,
+
+    x = torch.tensor([[1, 0], [3, 2]])
+
+    call regular `nonzero()` on x will give you a tensor `tensor([[0, 0], [1, 0], [1, 1])`
+
+    call `nonzero_static(x, size=4)` on x will give you a tensor
+    `tensor([[0, 0], [1, 0], [1, 1], [fill_value, fill_value])` (padded)
+
+    call `nonzero_static(x, size=2)` on x will give you a tensor
+    `tensor([[0, 0], [1, 0])` (truncated)
+    """
+    # Check if 'fill_value' is an integer type
+    assert isinstance(fill_value, int), "'fill_value' must be an integer"
```

Comment:
Use check from prim_utils rather than assert for user checks

Reply:
Which prim_utils? I don't see any match when I searched "prim_utils". By looking at `torch/_prims/__init__.py`, it looks like many ops there are actually using `assert`.